### PR TITLE
Add an end to end test for helix gateway

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#2821 
### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This change add e2e test and fix some bugs

### Tests

- [X] The following tests are written for this issue:

TestFilePullChannelE2E

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public static GatewayServiceEvent translateCurrentStateChangeToEvent(String inst
    * @param shardStateMap the initial state of shards on the participant. Could be empty map
    * @return
    */
-  public static GatewayServiceEvent translateCurrentStateDiffToInitConnectEvent(String instanceName, String clusterName,
+  public static GatewayServiceEvent translateCurrentStateDiffToInitConnectEvent( String clusterName, String instanceName,
```
Comment (zpinto) [inline helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java:168]: Fix whitespace before "String"

Comment (zpinto) [reply helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java:168]: Same for above methods

**Review by zpinto — COMMENTED** [review]
Overall LGTM, comments are for formatting.

File: helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public static GatewayServiceEvent translateClientCloseToEvent(String instanceNam
    * @param shardStateMap
    * @return
    */
-  public static GatewayServiceEvent translateCurrentStateChangeToEvent(String instanceName, String clusterName,
+  public static GatewayServiceEvent translateCurrentStateChangeToEvent( String clusterName, String instanceName,
```
Comment (zpinto) [inline helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java:147]: space before String

File: helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java [hunk 3]
```diff
@@ -130,7 +130,7 @@ public static GatewayServiceEvent translateShardStateMessageToEventAndUpdateCach
    * @param clusterName the cluster name
    * @return GatewayServiceEvent
    */
-  public static GatewayServiceEvent translateClientCloseToEvent(String instanceName, String clusterName) {
+  public static GatewayServiceEvent translateClientCloseToEvent( String clusterName, String instanceName) {
```
Comment (zpinto) [inline helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java:133]: Space before String

**Review by junkaixue — APPROVED** [review]

**Comment by xyuanlu** [comment]
This change is ready to be merged. Approved by @junkaixue 
Commit message:
Add an end to end test for helix gateway

**Pull request merged**